File: falena/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace falena.Models
{
    public class Product
    {
        [Required]
        [Key]

        public int Id { get; set; }
        [Required]
        [MaxLength (50)]
        public string Name { get; set; }
        [Required]
        public double  PriceList { get; set; }
        [DisplayName("Expected revenue")]
        public double ExpectedRevenue { get; set; }
        [DisplayName("Final price")]
        public double FinalPrice => PriceList + ExpectedRevenue;

        [DisplayName("Current stock")]
        public int CurrentStock { get; set; }

        public bool IsActive { get; set; }

    }
}

File: falena/wwwroot/js/Product.js


function updateFinalPrice() {
    var priceList = parseInt($("#PriceList").val()) + 0;
    var expectedRevenew = parseInt($("#ExpectedRevenue").val()) + 0;
    if (priceList > 0 && expectedRevenew > 0) {
        $("#FinalPrice").val(priceList + expectedRevenew);
    }
}

function updateCurrentStock() {
    var purchases = parseInt($("#Purchases").val());
    var sales = parseInt($("#Sales").val());
    if (purchases > 0 && sales > 0) {
        $("#CurrentStock").val(purchases - sales);
    }

}